src/client/ds/object_meta.cc

```diff
@@ -347,9 +363,15 @@ void ObjectMeta::SetMetaData(ClientBase* client, const json& meta) {
     ObjectID member_id =
         ObjectIDFromString(tree["id"].get_ref<std::string const&>());
     if (IsBlob(member_id)) {
-      if (client_ == nullptr /* traverse to account blobs */ ||
-          tree["instance_id"].get<InstanceID>() == client_->instance_id()) {
+      if (client_ == nullptr) {
         VINEYARD_CHECK_OK(buffer_set_->EmplaceBuffer(member_id));
+      } else {
+        std::shared_ptr<struct InstanceStatus> instance_status = nullptr;
+        VINEYARD_CHECK_OK(client_->InstanceStatus(instance_status));
+        if (tree["instance_id"].get<InstanceID>() ==
+            instance_status->instance_id) {
+          VINEYARD_CHECK_OK(buffer_set_->EmplaceBuffer(member_id));
```

Comment:
You would better cache the instance_status in RPCClient as a member, otherwise you would need call it for so many times when building/resolving metadata....